include/picongpu/particles/Particles.tpp

```diff
@@ -48,12 +48,144 @@
 #include <iostream>
 #include <limits>
 #include <memory>
-
+#include <utility>
 
 namespace picongpu
 {
     using namespace pmacc;
 
+    namespace detail
+    {
+        /* Helper to check if a member exists
+         *
+         * Derived from C++17 std::void_t.
+         * This implementation will be removed with Void provided by alpaka 0.6.0 release (not included in the 0.6.0rc3
+         * we currently using).
+         */
+        template<class...>
+        using Void = void;
+
+        /** Calculate the scaling factor for each direction.
+         *
+         * The scaling factor is derived from the reference size of the local domain and a scaling factor provided by
+         * the user.
+         *
+         * @tparam T_ExchangeMemCfg exchange configuration for a species
+         * @tparam T_Sfinae Type for conditionally specialization (no input parameter)
+         * @{
+         */
+        template<typename T_ExchangeMemCfg, typename T_Sfinae = void>
+        struct DirScalingFactor
+        {
+            //! @return factor to scale the amount of memory for each direction
+            static floatD_64 get()
+            {
+                return floatD_64::create(1.0);
+            }
+        };
+
+        /** Specialization for species with exchange memory information which provides
+         * DIR_SCALING_FACTOR and REF_LOCAL_DOM_SIZE
+         */
+        template<typename T_ExchangeMemCfg>
+        struct DirScalingFactor<
+            T_ExchangeMemCfg,
+            Void<
+                decltype(std::declval<T_ExchangeMemCfg>().DIR_SCALING_FACTOR),
+                typename T_ExchangeMemCfg::REF_LOCAL_DOM_SIZE>>
+        {
```

Comment:
I see you are using the member check from alpaka, nice.

Reply:
Yes, and I will switch to the alpaka version as we use the latest alpaka.

Reply:
btw: this feature is very easy to use, I love it :heart: